jenkins_node_scanner.py

```diff
@@ -0,0 +1,157 @@
+        raise
+    except:
+        logging.exception('Unexpected error, continuing anyway.')
+
+
+def get_args():
+    """Configure arguments and parse them."""
+    parser = argparse.ArgumentParser(
+        description='Write Jenkins nodes to a file.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('url',
+                        help='Full URL to the Jenkins master, such as http://jenkins.')
+    parser.add_argument('output_file',
+                        help='Filename to write the node list.')
+    parser.add_argument('--username',
+                        help='Jenkins username.')
+    parser.add_argument('--password',
+                        help='Jenkins password.')
+    parser.add_argument('--exclude-regex', default='^master$',
+                        help='Exclude any nodes matching this regex.')
+    parser.add_argument('--timeout', default=60, type=int,
+                        help='Timeout in seconds for the Jenkins call.')
+    parser.add_argument('--period', default=60, type=int,
+                        help='How many seconds to wait between scans.')
+    parser.add_argument('--prometheus-port', default=8000, type=int,
+                        help='Port on which to expose prometheus metrics.')
+    parser.add_argument('--target-port', default=9100, type=int,
+                        help='Target port to be scraped (e.g. 9100 for node_exporter).')
+    parser.add_argument('-v', '--verbose', action='store_true',
+                        help='Enable more detailed logging.')
+    return parser.parse_args()
+
+
+def get_node_ip(master, node):
+    """Returns the ip of a node."""
+    with JENKINS_API_EXCEPTIONS.labels('get_node_config').count_exceptions():
+        with JENKINS_API_LATENCY.labels('get_node_config').time():
+            node_config = master.get_node_config(node['name'])
+    if 'hudson.plugins.swarm.SwarmSlave' not in node_config:
+        logging.error('Unrecognized node type: %s', node['name'])
+        return None
+
+    tree = ElementTree.fromstring(node_config)
+    description = tree.find('description').text
+    # Parses "Swarm slave from 10.1.1.1 : null" (after : can be arbitrary text)
+    return description.split('from ')[1].split(' :')[0]
+
+
+def get_nodes(master):
+    """Get a list of nodes from a Jenkins master."""
+    logging.debug('Connecting to master')
+    with JENKINS_API_EXCEPTIONS.labels('get_nodes').count_exceptions():
+        with JENKINS_API_LATENCY.labels('get_nodes').time():
+            return master.get_nodes()
+
+
+def main():
+    """Run the main scanning loop."""
+    args = get_args()
+    logging.basicConfig(level=logging.INFO)
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+
+    logging.info('Connecting to Jenkins master at %s', args.url)
+    master = jenkins.Jenkins(
+        args.url,
+        username=args.username,
+        password=args.password,
+        timeout=args.timeout)
```

Comment:
I am certainly no master of python style, but shouldn't this line have a trailing comma and the close-parenthesis be on the next line?